clients/src/test/java/org/apache/kafka/common/network/SSLSelectorTest.java

```diff
@@ -273,4 +320,84 @@ private void sendAndReceive(String node, String requestPrefix, int startIndex, i
         }
     }
 
+    private void createSelector(final int netReadBufSize, final int netWriteBufSize, final int appBufSize) {
+        
+        if (this.selector != null)
+            this.selector.close();
+        this.channelBuilder = new SSLChannelBuilder(SSLFactory.Mode.CLIENT) {
+
+            @Override
+            protected SSLTransportLayer buildTransportLayer(SSLFactory sslFactory, String id, SelectionKey key) throws IOException {
+                SocketChannel socketChannel = (SocketChannel) key.channel();
+                SSLEngine sslEngine = sslFactory.createSSLEngine(socketChannel.socket().getInetAddress().getHostName(),
+                                socketChannel.socket().getPort());
+                return new TestSSLTransportLayer(id, key, sslEngine, netReadBufSize, netWriteBufSize, appBufSize);
+            }
+
+
+        };
+        this.channelBuilder.configure(sslClientConfigs);
+        this.selector = new Selector(5000, new Metrics(), new MockTime(), "MetricGroup", new LinkedHashMap<String, String>(), channelBuilder);
+    }
+
+    /**
+     * SSLTransportLayer with overrides for packet and application buffer size to test buffer resize
+     * code path. The overridden buffer size starts with a small value and increases in size when the buffer
+     * size is retrieved to handle overflow/underflow, until the actual session buffer size is reached.
+     */
+    private static class TestSSLTransportLayer extends SSLTransportLayer {
+
+        private int netReadBufSizeOverride;
+        private int netWriteBufSizeOverride;
+        private int appBufSizeOverride;
+        private boolean isInitialized;
+
+        public TestSSLTransportLayer(String channelId, SelectionKey key, SSLEngine sslEngine, 
+                int netReadBufSize, int netWriteBufSize, int appBufSize) throws IOException {
+            super(channelId, key, sslEngine);
+            this.netReadBufSizeOverride = netReadBufSize;
+            this.netWriteBufSizeOverride = netWriteBufSize;
+            this.appBufSizeOverride = appBufSize;
+            this.isInitialized = true;
+        }
+        
+        @Override
+        protected int netReadBufferSize() {
+            int size = super.netReadBufferSize();
+            if (!isInitialized) {
+                size = 10;
+            } else if (netReadBufSizeOverride != 0) {
```

Comment:
Would it be better to use `null` for the no override case instead of `0`? I think it's clearer when reading the test code (at first I thought `0` meant an empty buffer.